### PR TITLE
Require protocol when validating URLs

### PR DESCRIPTION
**Pull request by fquffio**
This PR fixes an issue with fields of type URL: the validator did _not_ require protocol to be present. Thus, `www.example.com` would have been a valid value for a **Profile**'s `website` or an **AuthProvider**'s `url`.

Now, URL fields are required to include a protocol.

Instead of using `Validator::urlWithProtocol()` I've explicitly added the rule for the field. This is required to avoid a breaking change that would have caused some clients to not recognize that type of error anymore: if `Validator::urlWithProtocol` was used, a failure due to an invalid URL would have resulted in a validation error for rule `website.urlWithProtocol`, which differs from the rule `website.url` we have now. Using `Validator::add()` I was able to explicitly set the name of the rule.

### CODE REVIEW

File: plugins/BEdita/Core/tests/TestCase/Model/Entity/ProfileTest.php [hunk 1]
```diff
@@ -88,4 +90,51 @@ public function testAccessible()
...
+        ];
+    }
+
+    /**
+     * Test that Website URL is correctly standardized.
+     *
+     * @param mixed $expected Expected result.
+     * @param mixed $website Website value.
+     * @return void
+     *
```
Comment (stefanorosanelli) [inline plugins/BEdita/Core/tests/TestCase/Model/Entity/ProfileTest.php]: [super minor] remove empty lines between annotations (Cake 4 cs rules)

**Review by stefanorosanelli — APPROVED** [review]

File: plugins/BEdita/Core/tests/TestCase/Model/Entity/ProfileTest.php (hunk 1 above)
Comment (fquffio) [reply plugins/BEdita/Core/tests/TestCase/Model/Entity/ProfileTest.php]:
```suggestion
```

File: plugins/BEdita/Core/tests/TestCase/Model/Table/AuthProvidersTableTest.php [hunk 2]
```diff
@@ -88,15 +89,17 @@ public function validationProvider()
...
                 [
                     'name' => 'some_unique_value',
                     'url' => 'https://example.com/oauth2',
                     'params' => [],
                 ],
             ],
             'notUnique' => [
-                false,
+                [
+                    'name.unique',
```
Comment (stefanorosanelli) [inline plugins/BEdita/Core/tests/TestCase/Model/Table/AuthProvidersTableTest.php:102]: Now we are using `$authProvider = $this->AuthProviders->newEntity([]);` in this test case (Cake 4 compatibility) validation output differs

```suggestion
                    'name._required'
                    'name.unique'
```

File: plugins/BEdita/Core/tests/TestCase/Model/Table/AuthProvidersTableTest.php [hunk 3]
```diff
@@ -88,15 +89,18 @@ public function validationProvider()
...
                     'name' => 'some_unique_value',
                     'url' => 'https://example.com/oauth2',
                     'params' => [],
                 ],
             ],
             'notUnique' => [
-                false,
+                [
+                    'name._required'
+                    'name.unique'
```
Comment (fquffio) [inline plugins/BEdita/Core/tests/TestCase/Model/Table/AuthProvidersTableTest.php]:
```suggestion
                    'name._required',
                    'name.unique',
```

**Pull request merged**